modules/post/linux/gather/checkvm.rb

```diff
@@ -112,10 +114,21 @@ def run
       end
     end
 
-    # Xen bus check
     if not vm
-      if cmd_exec("ls -1 /sys/bus").to_s.split("\n").include?("xen")
-        vm = "Xen"
+      product_name = read_file('/sys/class/dmi/id/product_name')
+      if product_name
+        case product_name.gsub("\n", " ")
+      	when /vmware/i
+      	  vm = "VMware"
+      	when /virtualbox/i
+      	  vm = "VirtualBox"
+      	when /xen/i
+      	  vm = "Xen"
+      	when /KVM/i
+      	  vm = "KVM"
+      	when /oracle/i
+      	  vm = "Oracle Corporation"
+      	end
```

Comment:
As noted by msftidy ensure use of spaces for indention.

```
--- Checking new and changed module syntax with tools/dev/msftidy.rb ---
modules/post/linux/gather/checkvm.rb - [INFO] No CVE references found. Please check before you land!
modules/post/linux/gather/checkvm.rb:121 - [WARNING] Space-Tab mixed indent: " \twhen /vmware/i\n"
modules/post/linux/gather/checkvm.rb:122 - [WARNING] Space-Tab mixed indent: " \t vm = \"VMware\"\n"
modules/post/linux/gather/checkvm.rb:123 - [WARNING] Space-Tab mixed indent: " \twhen /virtualbox/i\n"
modules/post/linux/gather/checkvm.rb:124 - [WARNING] Space-Tab mixed indent: " \t vm = \"VirtualBox\"\n"
modules/post/linux/gather/checkvm.rb:125 - [WARNING] Space-Tab mixed indent: " \twhen /xen/i\n"
modules/post/linux/gather/checkvm.rb:126 - [WARNING] Space-Tab mixed indent: " \t vm = \"Xen\"\n"
modules/post/linux/gather/checkvm.rb:127 - [WARNING] Space-Tab mixed indent: " \twhen /KVM/i\n"
modules/post/linux/gather/checkvm.rb:128 - [WARNING] Space-Tab mixed indent: " \t vm = \"KVM\"\n"
modules/post/linux/gather/checkvm.rb:129 - [WARNING] Space-Tab mixed indent: " \twhen /oracle/i\n"
modules/post/linux/gather/checkvm.rb:130 - [WARNING] Space-Tab mixed indent: " \t vm = \"Oracle Corporation\"\n"
modules/post/linux/gather/checkvm.rb:131 - [WARNING] Space-Tab mixed indent: " \tend\n"
```

Reply:
Nice push right as I made this comment 👍

Reply:
:smile: